src/features/account/AccountDeleteVerificationCodeModal.tsx

```diff
@@ -0,0 +1,97 @@
+import React from 'react';
+
+import {
+  Modal,
+  ModalBody,
+  ModalCloseButton,
+  ModalContent,
+  ModalOverlay,
+} from '@chakra-ui/react';
+import { zodResolver } from '@hookform/resolvers/zod';
+import { useRouter } from 'next/navigation';
+import { parseAsInteger, parseAsString, useQueryStates } from 'nuqs';
+import { SubmitHandler, useForm } from 'react-hook-form';
+import { useTranslation } from 'react-i18next';
+
+import { Form } from '@/components/Form';
+import {
+  VerificationCodeForm,
+  useOnVerificationCodeError,
+} from '@/features/auth/VerificationCodeForm';
+import { trpc } from '@/lib/trpc/client';
+
+import {
+  FormFieldsVerificationCode,
+  zFormFieldsVerificationCode,
+} from '../auth/schemas';
+
+export const SEARCH_PARAM_VERIFY_EMAIL = 'delete-account';
+
+export const AccountDeleteVerificationCodeModale = () => {
+  const { t } = useTranslation(['account']);
+  const router = useRouter();
+  const [searchParams, setSearchParams] = useQueryStates(
+    {
+      [SEARCH_PARAM_VERIFY_EMAIL]: parseAsString,
+      token: parseAsString,
+      attempts: parseAsInteger,
+    },
+    {
+      history: 'replace',
+    }
+  );
+
+  const onClose = () => {
+    setSearchParams({
+      [SEARCH_PARAM_VERIFY_EMAIL]: null,
+      token: null,
+      attempts: null,
+    });
+  };
+
+  const form = useForm<{ code: string }>({
```

Comment:
We don't infer from the zod schema here?

Reply:
Yes we should, I will fix that